### DISCUSSION TITLE
Yellow and Red symbols over tables in TOM Explorer

### DISCUSSION THREAD

**Discussion by emzwurld0911** [TabularEditor/TabularEditor3#1331] [🙋♀️TE3 Q&A]
Hi
I'm just learning about Tabular Editor 3 and Power BI modeling. I've brought in my vertipaq file and am looking at an inherited database so I can try to understand the relationships, however I see lots of yellow triangles and red x's over many tables. Can anyone help me understand what these are and how I should interpret them?

**Comment by otykier** [comment]
Hi @emzwurld0911 

Yellow triangles generally mean that there are one or more **warnings** in the DAX expressions of the object. For example, in the screenshot below, I am editing the DAX expression of a *calculated column*. This expression refers to other columns of the table. In DAX, it is recommended to always [fully qualify a column reference with a table name](https://learn.microsoft.com/en-us/dax/best-practices/dax-column-measure-references#columns), but since I only wrote the column name *without* the table qualifier, I get a warning:

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/924f6732-f2c2-4a45-a651-4f3717aef729)

In general, DAX expressions containing one or more warnings can still be evaluated and return a result. However, the highlighted issue could be ambiguous causing a wrong or unexpected result to be produced.

Red x's mean that the DAX expressions have one or more **errors**, which would prevent the expression from returning a result. For example, in the following measure, I deliberately misspelled the name of another measure in the model, causing a semantic error:

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/bfeb35f9-70cd-4a22-a345-a521ea259f95)

If I tried to add this measure to a visual in Power BI, I would also get an error.

If you see smaller circles in the **TOM Explorer**, it simply means that one or more objects within that object has an error, so you'd need to expand the object to see the full list:

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/a295583e-3d3a-46d0-a1fd-cd4123dda1ae)

Alternatively, you can view a complete list of issues in the **Messages** view. The items on the list correspond to yellow triangles or red circles in the TOM Explorer. Double-clicking on an item in the list will take you directly to that issue:

![image](https://github.com/TabularEditor/TabularEditor3/assets/8976200/8ee3128c-12d8-4e75-b737-8587e0a27629)

Hope this helps!

**Comment by emzwurld0911** [comment]
This is perfect. Thank you so much for the explanation.

Emma
